### PR TITLE
Add new content to salt.md and proofread all documentation.

### PR DESCRIPTION
**Pull request by sliute**
I also found a broken link in pillar/README.md and created an empty README.md file in ../../_vagrant/salt/pillar where I assumed that content should exist.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,6 @@
 
 # GitBook
 /_book/
+
+# PHPStorm
+/.idea/
```
Comment (LukeCarrier) [inline .gitignore]: Could these files be [added to your global excludes file](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore) instead?  I try to keep project `.gitignore`s separate from development tools.

File: _docs/README.md [hunk 2]
```diff
@@ -5,9 +5,9 @@ Ubiquitous is a multi-purpose Moodle hosting environment.
 It fulfils three requirements:
 
 1. _Provisioning_ of machines with either a container or virtualisation platform.
-    * In local development environments we manage [VirtualBox](https://www.virtualbox.org/) virtual machines with [Vagrant](https://www.vagrantup.com/)
+    * In local development environments we manage [VirtualBox](https://www.virtualbox.org/) virtual machines with [Vagrant](https://www.vagrantup.com/) (version [1.9.6](https://releases.hashicorp.com/vagrant/1.9.6/) for now, because of [this bug](https://github.com/mitchellh/vagrant/issues/8770)).
```
Comment (LukeCarrier) [inline _docs/README.md]: Nice touch, hopefully 1.9.8 will be released soon :)

File: _docs/getting-started/README.md [hunk 3]
```diff
@@ -39,11 +39,11 @@ The following machines are provided for testing production configurations:
 * A GoCD instance for testing deployment pipelines
 * A nameserver running BIND for testing your DNS configuration
 
-Machines started and stopped independently of one another and we configure handy groups to simplify this process. You're also able to add your own machines as required.
+Machines start and stop independently of one another and we configured handy groups to simplify this process. You're also able to add your own machines as required.
```
Comment (LukeCarrier) [inline _docs/getting-started/README.md]: Can we rephrase this to:

> Machines can be started and stopped independently or using Vagrant groups. You're also able to add your own machines as required.

File: _docs/getting-started/development.md [hunk 4]
```diff
@@ -74,7 +74,7 @@ The above may take some time to complete. Once the above commands complete, the
 * [Behat instance](http://192.168.120.50/behat/) --- your development environment's Behat `wwwroot`
 * [Behat fail dump](http://192.168.120.50/data/behat-faildump/) --- screenshots and page snapshots for failing Behat tests
 * [MailCatcher](http://192.168.120.200:1080/) --- a simple mail server that allows you to browse all of the email it receives
-* PostgreSQL --- `192.168.120.150:5432`
+* PostgreSQL database at `192.168.120.150:5432` --- use a tool like [DBeaver](http://dbeaver.jkiss.org/) to connect to it.
```
Comment (LukeCarrier) [inline _docs/getting-started/development.md]: I'd like to avoid recommending specific tools here. Could we link to a [list like this](https://wiki.postgresql.org/wiki/Community_Guide_to_PostgreSQL_GUI_Tools) instead?

File: _docs/getting-started/development.md [hunk 5]
```diff
@@ -246,16 +246,16 @@ $ vagrant ssh salt --command "sudo salt -G 'group:selenium' state.apply"
 Once complete, the following services will be available to you:
 
 * [Selenium Grid console](http://192.168.120.100:4444/grid/console) --- see an overview of available nodes, helpful for diagnosing registration issues
-* VNC for the Selenium Chrome node --- `192.168.120.105:5999`
-* VNC for the Selenium Firefox node --- `192.168.120.110:5999`
+* VNC for the Selenium Chrome node at `192.168.120.105:5999` --- use a tool like [Remmina](https://www.remmina.org/wp/) to connect to it
```
Comment (LukeCarrier) [inline _docs/getting-started/development.md]: Again, maybe worth linking to [something like this](https://en.wikipedia.org/wiki/Comparison_of_remote_desktop_software)?

File: _docs/README.md [hunk 6]
```diff
@@ -5,9 +5,9 @@ Ubiquitous is a multi-purpose Moodle hosting environment.
...
     * Continuous integration platforms run on purpose-built [Docker](https://www.docker.com/) containers
```
Comment (LukeCarrier) [inline _docs/README.md]: If we're adding full stops to some list items, can we add them to all of them?

File: _docs/pillar/README.md [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 [The Salt pillar](https://docs.saltstack.com/en/latest/topics/tutorials/pillar.html) provides a home for configuration data that doesn't belong within the Salt state tree, enabling developers to create more reusable states.
 
-This document provides an overview of the available pillar values and where they're used. A complete sample of the pillar tree, with relevant top file, can be found in the [Vagrant configuration](../vagrant/salt/pillar).
+This document provides an overview of the available pillar values and where they're used. A complete sample of the pillar tree, with relevant top file, can be found in the [Vagrant configuration](../../_vagrant/salt/pillar/README.md).
```
Comment (LukeCarrier) [inline _docs/pillar/README.md]: My intention in linking here was to show developers the source code. I suggest we either:

1. Link to GitHub directly, so that the folder listing is visible; or
2. Remove this link entirely.

File: _docs/roles/README.md [hunk 8]
```diff
@@ -17,46 +17,44 @@ In Ubiquitous, a role is an individual Salt state with a corresponding Salt grai
...
+| Dependencies | Dependants |
 | --- | --- |
-| Dependencies | `app` |
-| Dependants | None |
+| `app` | None |
 
 This role augments the `app` role with easy access to Behat's faildump and enables debugging and profiling using Xdebug.
 
-The Behat faildump for each platform can be accessed from a browser at `/data/behat-faildump`.
+The Behat faildump for each platform can be accessed from a browser at `{platform IP}/data/behat-faildump`.
```
Comment (LukeCarrier) [inline _docs/roles/README.md]: Could we go with `{document root}` here?

File: _docs/roles/app.md [hunk 9]
```diff
@@ -46,6 +46,9 @@ $ salt <minion ID> pillar.keys platforms
 
 The `app-gocd-agent` role writes a number of scripts facilitating the deployment process to `/usr/local/ubiquitous/bin`:
 
-* `ubiquitous-info` helps administrators understand the running state of a platform by describing the current configuration, examining the three symbolic links (the nginx `server {}` entry, the PHP-FPM pool and the `current` symlink in the user's home directory)
-* `ubiquitous-install-release` copies new source files into a release directory ready for switching over
-* `ubiquitous-set-current-release` changes the active release by enabling the inactive FPM pool before altering each of the symbolic links and reloading the services as necessary
+* `ubiquitous-info` helps administrators understand the running state of a platform. It describes the current configuration by examining the three symbolic links
+    * (the nginx `server {}` entry
```
Comment (LukeCarrier) [inline _docs/roles/app.md]: The formatting here is a little off. Maybe something like...

> * `ubiquitous-info` helps administrators understand the running state of a platform by describing the current configuration. It does by examining three symbolic links:
>     * the active nginx `server {}` entry in `/etc/nginx/sites-enabled.d`;
>     * the active PHP-FPM pool entry in `/etc/php-fpm/pools-enabled.d`; and
>     * the `current` symlink in the user's home directory

File: _docs/roles/gocd.md [hunk 10]
```diff
@@ -13,7 +13,7 @@ If you see messages to the effect of the following in `go-agent-launcher.log` (`
 INFO   | jvm 1    | 2017/04/12 17:36:37 | javax.net.ssl.SSLHandshakeException: Remote host closed connection during handshake
 ```
 
-The likely cause is that the GoCD server's SSL certificate changed (e.g. because the GoCD server has been rebuilt) and the agents have the old certificate cached.
+the likely cause is that the GoCD server's SSL certificate changed (e.g. because the GoCD server has been rebuilt) and the agents have the old certificate cached.
```
Comment (LukeCarrier) [inline _docs/roles/gocd.md:16]: Is this grammatically correct? It feels questionable :thinking:

File: _vagrant/salt/pillar/README.md [hunk 11]
```diff
@@ -0,0 +1 @@
+A complete sample of the pillar tree should be added here, as promised.
```
Comment (LukeCarrier) [inline _vagrant/salt/pillar/README.md]: As above, we can remove this file as GitBook will never process it (being outside of the `_docs` directory). We'll either remove the corresponding link or point it directly at GitHub so we're able to show the directory listing.

**Review by LukeCarrier — CHANGES_REQUESTED** [review]
Some nice polish added here -- thanks for taking the time to go through all of the documentation.

My feedback falls into three areas:

1. `.gitignore`s should be project specific and I try to avoid putting IDE and editor configurations in there.
2. On tool selections (e.g. Dbeaver and Remmina) I'm happy to make cross-platform, open source suggestions or link to lists of similar tools?
3. I'm not concerned about formatting of lists provided we're consistent -- I'm aware that's not currently the case. Full stop at the beginning and full stop at the end of each list item sound good?

File: .gitignore (hunk 1 above)
Comment (sliute) [reply .gitignore]: Agree. Took them out of project .gitignore.

File: _docs/README.md (hunk 6 above)
Comment (sliute) [reply _docs/README.md]: Finally, someone who cares about this! Here's how I view list formatting:
- if list elements are not full sentences (i.e. they lack predicates), only the last of them gets a period at the end, since the list is pretty much like a standard comma-based one and its enclosing sentence has begun just above the list.
- if list elements contain predicates, each gets a period, since they are full sentences.
-  we should avoid mixed lists, i.e. lists that mix both of the above.
The latest commit uses these rules - see if you like how it looks.

File: _docs/getting-started/README.md (hunk 3 above)
Comment (sliute) [reply _docs/getting-started/README.md]: Done.

File: _docs/getting-started/development.md (hunk 4 above)
Comment (sliute) [reply _docs/getting-started/development.md]: Done

File: _docs/getting-started/development.md (hunk 5 above)
Comment (sliute) [reply _docs/getting-started/development.md]: Done

File: _docs/pillar/README.md (hunk 7 above)
Comment (sliute) [reply _docs/pillar/README.md]: Linked to GitHub.

File: _docs/roles/README.md (hunk 8 above)
Comment (sliute) [reply _docs/roles/README.md]: Done.

File: _docs/roles/app.md (hunk 9 above)
Comment (sliute) [reply _docs/roles/app.md]: Done.

File: _docs/roles/gocd.md (hunk 10 above)
Comment (sliute) [reply _docs/roles/gocd.md:16]: Yes, I think it is. Our code excerpt is simply an item in a sentence that continues when it ends.

File: _vagrant/salt/pillar/README.md (hunk 11 above)
Comment (sliute) [reply _vagrant/salt/pillar/README.md]: Done.

**Review by LukeCarrier — APPROVED** [review]
Hey @sliute,

Changes look good -- thanks. Please could you [squash these three commits down to one](https://davidwalsh.name/squash-commits-git) to keep the history clean, and just include `docs:` at the beginning of the commit message.

**Comment by sliute** [comment]
Hey @LukeCarrier - squashed last 3 commits into one. Went for [this way](https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git/5201642#5201642) of squashing.

Enjoy the holidays!

**Comment by LukeCarrier** [comment]
@sliute looks like the commit message got lost in translation :stuck_out_tongue: Would you be able to `git commit --amend` it before merging?

**Comment by sliute** [comment]
Mais biensur, c'est ready :-P

**Comment by LukeCarrier** [comment]
Merci beaucoup @sliute!
